### PR TITLE
Project search filtering

### PR DESCRIPTION
**Pull request by Farce4wrd**
-Implemented filtering by dropdown (Tech, Level, Difficulty, Duration, and Semester)
-Implemented filtering through search input in conjunction

### CODE REVIEW

File: src/content/test/mytest.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
```
Comment (cengique) [inline src/content/test/mytest.md:1]: @Farce4wrd do we need this file?

File: src/components/ProjectSearch.vue [hunk 2]
```diff
@@ -1,51 +1,168 @@
...
-    <section class="mbe40"> 
         
-        <Input  id="7" is-underlined is-underlined-with-background placeholder="Enter project name, student, technology…" 
-                label="Search for projects" type="text" v-model="search_text" />
+        <div>
+            <div>
+                <label>Semester</label>
+                <Select name="semester" unique-id="sem" @selected="(value) => { semester = value }" 
+                    label-copy="Select a semester to filter results" 
+                    :options="semesterList" :is-multiple="true" :multiple-size="3"></Select>
```
Comment (cengique) [inline src/components/ProjectSearch.vue:118]: @Farce4wrd let's make all of them multi-select

File: src/components/ProjectSearch.vue [hunk 3]
```diff
@@ -1,51 +1,168 @@
...
+        });
+        }else{
+            optionSet.add(arrayContainer.data[x]);
+        }
+        
+    });
+    return Array.from(optionSet).map(option =>({value:option, label:option}));
+}
+
+const levelOptions = computed(() => createOptions("levels") );
```
Comment (cengique) [inline src/components/ProjectSearch.vue:48]: Nice and clean

File: src/components/TAPHeader.vue [hunk 4]
```diff
@@ -17,6 +17,9 @@ import { Header, HeaderNav, HeaderNavItem, Menu } from "agnostic-vue";
         <HeaderNavItem>
           <a href="">Join the discussion</a>
         </HeaderNavItem>
+        <HeaderNavItem>
```
Comment (cengique) [inline src/components/TAPHeader.vue:20]: Did we need this?

**Review by cengique — CHANGES_REQUESTED** [review]
Minor changes requested and some pending questions.

File: src/content/test/mytest.md (hunk 1 above)
Comment (Farce4wrd) [reply src/content/test/mytest.md:1]: Oops! I had that when I was trying to learn and understand ASTRO. Nope, it'll have to go

File: src/components/TAPHeader.vue (hunk 4 above)
Comment (Farce4wrd) [reply src/components/TAPHeader.vue:20]: I assume this is related to the test.md. I'll remove it

**Review by cengique — APPROVED** [review]

**Pull request merged**